pkg/extensionController/controller.go

```diff
@@ -103,13 +103,16 @@ func (c *controllerImpl) requiredFilesAvailable(extension *extensionAPI.JsExtens
 
 func existsFileInBfs(bfsFiles []bfs.BfsFile, requiredFile extensionAPI.BucketFsUpload) bool {
 	for _, existingFile := range bfsFiles {
-		if requiredFile.BucketFsFilename == existingFile.Name && requiredFile.FileSize == existingFile.Size {
+		if requiredFile.BucketFsFilename == existingFile.Name && (requiredFile.FileSize < 0 || requiredFile.FileSize == existingFile.Size) {
```

Comment:
Maybe consider to move detection unspecified file size to a private method with speaking name?
```golan
func fileMatches(name) bool {
}
```